perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/FacilitiesManagerBlImpl.java

```diff
@@ -404,10 +404,19 @@ public PerunBl getPerunBl() {
 		return getFacilitiesManagerImpl().getAssignedFacilities(sess, securityTeam);
 	}
 
-	public List<Facility> getFacilitiesByAttribute(PerunSession sess, Attribute attribute) throws InternalErrorException, WrongAttributeAssignmentException {
-		getPerunBl().getAttributesManagerBl().checkNamespace(sess, attribute, AttributesManager.NS_FACILITY_ATTR);
-		if(!(getPerunBl().getAttributesManagerBl().isDefAttribute(sess, attribute) || getPerunBl().getAttributesManagerBl().isOptAttribute(sess, attribute))) throw new WrongAttributeAssignmentException("This method can process only def and opt attributes");
-		return getFacilitiesManagerImpl().getFacilitiesByAttribute(sess, attribute);
+
+	public List<Facility> getFacilitiesByAttribute(PerunSession sess, String attributeName, String attributeValue) throws InternalErrorException, WrongAttributeAssignmentException {
+		try {
+			AttributeDefinition attributeDef = getPerunBl().getAttributesManagerBl().getAttributeDefinition(sess, attributeName);
+			Attribute attribute = new Attribute(attributeDef);
```

Comment:
If it can't be accessed directly, try to move ``isLargeAttribute()`` method to some kind of Utils class (core/base). It can be static method, since it only check type of passed attribute.